compass_app/app/lib/ui/search_form/view_models/search_form_viewmodel.dart

```diff
@@ -24,55 +29,55 @@ class SearchFormViewModel extends ChangeNotifier {
   final _log = Logger('SearchFormViewModel');
   final ContinentRepository _continentRepository;
   final ItineraryConfigRepository _itineraryConfigRepository;
-  List<Continent> _continents = [];
-  String? _selectedContinent;
-  DateTimeRange? _dateRange;
-  int _guests = 0;
+
+  // Immutable data class that holds state the View needs to display
+  SearchFormState searchFormState = SearchFormState();
```

Comment:
This replaces all data properties used by the views.

Reply:
This reminds me of Flutter Bloc "Cubit" implementation, where you have a State object and a Cubit, and then you subscribe to changes for this Cubit, getting a new State object: https://bloclibrary.dev/bloc-concepts/#cubit

It doesn't fully break from MVVM, but has some impracticalities.

For example, each time you change a property, you need to copy the old state and modify the value. This operation does not update the UI, but for that you also need to call to `NotifyListeners()`. This is where a package solution like `Cubit` comes in handy, since the `emit(new state)` call would replace the old state and trigger a UI refresh.

I'd probably prefer to keep ViewModels as they are, which follows the standard MVVM design pattern, where properties are directly exposed in the ViewModel instead of in a State object.

But I think it is very interesting to show an example on how this could be refactored into Flutter Bloc, Cubit or something like MVC.